proposals/0000-migrate-to-androidx-alertdialog.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Migrate to androidx.appcompat.app.AlertDialog
+author:
+- Peter Abbondanzo
+date: 2024-05-08
+---
+
+# RFC0000: Migrate to androidx.appcompat.app.AlertDialog
+
+## Summary
+
+This proposal suggests migrating [all existing uses](https://github.com/search?q=repo%3Afacebook%2Freact-native+android.app.AlertDialog&type=code) of `android.app.AlertDialog` to `androidx.appcompat.app.AlertDialog` to add support library features. Currently, the two primary uses are in the `AlertFragment` and `DevSupportManagerBase` classes.
+
+## Motivation
+
+Over the years, the vanilla `android.app.AlertDialog` has presented a suite of issues related [dark mode](https://github.com/facebook/react-native/issues/31345), [sizing](https://github.com/facebook/react-native/issues/25881), [RTL support](https://github.com/facebook/react-native/issues/25948), and [layout](https://github.com/facebook/react-native/pull/29832), just to list a few. It's the [explicit recommendation of Android](https://developer.android.com/topic/libraries/support-library) to use the support library wherever possible.
+
+We already use support library components elsewhere to power components like `ReactTextView`, `ReactSwitch`, and `ReactActivity` itelf. Moreover, we make use of the `AppCompatDelegate` to apply theming to support components in our `AppearanceModule`.
+
+Migrating to `androidx.appcompat.app.AlertDialog` would mitigate most of the aforementioned issues as it implicitly enforces using an `AppCompat` theme and provides compatible dark mode support out of the box without having to ask developers to modify their setup.
+
+## Detailed design
+
+There are two primary changes to make: in `AlertFragment` and `DevSupportManagerBase`.
+
+The `AlertFragment` will have to be modified to use the `androidx.appcompat.app.AlertDialog` builder instead of `android.app.AlertDialog`'s builder. Since the class inherits from `DialogFragment`, it only needs to yield a `Dialog` type.
+
+Similarly, the `DevSupportManagerBase` will also have to make this change from `android.app.AlertDialog` to `androidx.appcompat.app.AlertDialog`.
+
+To support activities that do not have AppCompat support, we will need to perform runtime checks against the current activity. This is necessary to ensure nothing crashes, as `AppCompatDelegateImpl` throws an `IllegalStateException` if the current activity does not have an AppCompat theme.
+
+The check would look something like this:
+
+```kotlin
+fun isAppCompatTheme(activityContext: Context): Boolean {
```

Comment:
LGTM 👍 Maybe we can have this inside a `ThemeUtil` or `DialogUtil` class somewhere and have tests for it?

Reply:
Yep, that's a great idea